lib/rules.js

```diff
@@ -196,6 +248,97 @@ export const PRODUCT_RULE = defineRuleString('#a^#b_0 * ...', '#a^(#b_0 + ...)')
 // e.g. x^5 / x^3 -> x^(5 - 3)
 export const QUOTIENT_RULE = defineRuleString('#a^#p / #a^#q', '#a^(#p - #q)')
 
+// e.g. 3x^2 * 2x^2 -> (3 * 2)(x^2 * x^2)
+export const MULTIPLY_COEFFICIENTS = defineRule(
+    (node) => {
+        return canApplyRule(ADD_EXPONENT_OF_ONE, node) ? {node} : null
```

Comment:
I guess `ADD_EXPONENT_OF_ONE_HELPER` does what you want. `ADD_EXPONENT_OF_ONE` looks through the given expression and applies the helper where necessary.